libnetdata/dictionary/README.md

```diff
@@ -2,4 +2,146 @@
 -->
 
+# Dictionaries
 
+Netdata dictionaries associate a `name` with a `value`:
+
+- A `name` can be any string.
+- A `value` can be anything.
+
+Such a pair of a `name` and a `value` consists of an `item` or an `entry` in the dictionary.
+
+Dictionaries provide an interface to:
+
+- **Add** an item to the dictionary
+- **Get** an item from the dictionary (provided its `name`)
+- **Delete** an item from the dictionary (provided its `name`)
+- **Traverse** the list of items in the dictionary
+
+Dictionaries are **ordered**, meaning that the order they have been added is preserved while traversing them. The called may reverse this order by passing the flag `DICTIONARY_FLAG_ADD_IN_FRONT` when creating the dictionary.
+
+Dictionaries are extremely fast in all operations. They are indexing the keys with `JudyHS` and they utilize a double-linked-list for the traversal operations. Deletion is the most expensive operation, usually somewhat slower than insertion.
+
+## Creation and destruction
+
+Use `dictionary_create()` to create a dictionary.
+
+Use `dictionary_destroy()` to destroy a dictionary.
+
+## Locks
+
+The dictionary allows both **single-threaded** operation (no locks - faster) and **multi-threaded** operation utilizing a read-write lock.
+
+The default is **multi-threaded**. To enable **single-threaded** add `DICTIONARY_FLAG_SINGLE_THREADED` to the flags when creating the dictionary.
+
+## Hash table operations
+
+The dictionary supports the following operations supported by the hash table:
+
+- `dictionary_set()` to add an item to the dictionary, or change its value.
+- `dictionary_get()` to get an item from the dictionary.
+- `dictionary_del()` to delete an item from the dictionary.
+
+### dictionary_set()
+
+This call is used to:
+
+- **add** an item to the dictionary.
+- **reset** the value of an existing item in the dictionary.
+
+If **resetting** is not desired, add `DICTIONARY_FLAG_DONT_OVERWRITE_VALUE` to the flags when creating the dictionary. In this case, `dictionary_set()` will return the value of the original item found in the dictionary instead of resetting it and the value passed to the call will be ignored.
+
+In **clone** mode for names, the call `dictionary_set_with_name_ptr()` can also return a pointer to the newly allocated name. This pointer will be set to the user supplied name even if the dictionary operated in **link** mode for names. 
+
+For **multi-threaded** operation, the `dictionary_set()` calls get an exclusive write lock on the dictionary.
+
+> **IMPORTANT**<br/>The above 2 function calls are also available in **unsafe** versions (without locks). These are to be used when traversing the dictionary. They should never be called without an active lock on the dictionary, which can only be acquired while traversing 
+
+### dictionary_get()
+
+This call is used to get the value of an item, given its name. It utilizes the JudyHS hash table for making the lookup.
```

Comment:
Right now we have two dictionary versions, but when an user read the previous sentence, they can think that we will never use `avl`, is this what we really want? 

Of course when `Judy` building issue is fixed, this will become truth, but I think it is better to push a commit to adjust the documentation when we are only using `Judy`.